pkg/cloudprovider/aws/instance.go

```diff
@@ -70,15 +70,15 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	if err != nil {
 		return nil, fmt.Errorf("creating fleet %w", err)
 	}
-	// TODO aggregate errors
-	if count := len(createFleetOutput.Errors); count > 0 {
-		return nil, fmt.Errorf("errors while creating fleet, %v", createFleetOutput.Errors)
-	}
 	if count := len(createFleetOutput.Instances); count != 1 {
-		return nil, fmt.Errorf("expected 1 instance, but got %d", count)
+		return nil, fmt.Errorf("expected 1 instance, but got %d due to errors %v", count, createFleetOutput.Errors)
```

Comment:
I think it would be good to have tests that simulate these cases ... since a lot rides on getting this right (both leaking instances and failing to launch them are pretty severe errors).

Reply:
Agreed, I think we're lacking some module level test. I can try adding one for the instance provider in this PR or can do in a follow up.

Reply:
Can do in follow up